lib/Pipeline/LLVMContainer.cpp

```diff
@@ -299,14 +301,20 @@ llvm::Error LLVMContainer::extractOne(llvm::raw_ostream &OS,
 }
 
 llvm::Error LLVMContainer::serialize(llvm::raw_ostream &OS) const {
-  getModule().print(OS, nullptr);
-  OS.flush();
+  ZstdCompressedOstream CompressedOS(OS, 3);
+  llvm::WriteBitcodeToFile(getModule(), CompressedOS);
+  CompressedOS.flush();
   return llvm::Error::success();
 }
 
 llvm::Error LLVMContainer::deserialize(const llvm::MemoryBuffer &Buffer) {
+  llvm::SmallVector<char> DecompressedData = zstdDecompress(Buffer);
+  llvm::MemoryBufferRef Ref{
+    { DecompressedData.data(), DecompressedData.size() }, "input"
+  };
+
   llvm::SMDiagnostic Error;
-  auto M = llvm::parseIR(Buffer, Error, Module->getContext());
+  auto M = llvm::parseIR(Ref, Error, Module->getContext());
```

Comment:
Is there a primitive for bitcode-files only?

Reply:
Switched to the bitcode-only version